### ISSUE TITLE
When customer changes default payment method, `Update your [other] subscriptions` feature doesn't work 

### ISSUE THREAD

**Issue by malithsen** [Automattic/woocommerce-payments#3235]
### Describe the bug
When the default payment method is updated, a prompt is shown to propagate the payment method update to existing subscriptions. However, this does not update the payment methods of existing subscriptions

![Screen Shot 2021-10-20 at 3 50 32 PM](https://user-images.githubusercontent.com/6216000/138170161-6d8d3332-495c-4926-9b07-95bbf35d3094.png)

### To Reproduce

1. As a customer, purchase a subscription with a test payment method
2. Purchase another subscription with a different payment method
2. Go to My Account -> Payment Methods and select "Make default" on a payment method
2. Click "Yes" in the "Would you like to update your subscriptions to use this new payment method - xxxx ending in xxxx (expires xx/xx)?"
3. Go to My Account -> Subscriptions
4. Notice that payment methods associated with the subscriptions are unchanged

### Actual behavior
Similar to WooCommerce Subscriptions, clicking "Yes" on "Would you like to update your subscriptions to use this new payment method" prompt should update existing subscriptions with the chosen default payment method

### Expected behavior
Subscriptions remain unchanged after changing the default payment method and agreeing to the subscription update prompt.

### Desktop:

* OS: macOS
* Browser: Chrome
* Version: 93.0

**Comment by haszari** [comment]
FYI @aprea - this is currently in scope for current sprint since there's work in progress. Can you (or someone in helix) take a look and decide whether we should proceed with current PR. If not, we could deprioritise this from current sprint to focus on improvements to empty state & subscription product create flow.

**Comment by haszari** [comment]
If fixing this feature is impractical, we could remove the misleading UI to make it consistent.

e.g. don't prompt: `Would you like to update your subscriptions to use this new payment method - xxxx ending in xxxx (expires xx/xx)?`

**Comment by leonardola** [comment]
I couldn't simulate the problem anymore. 

The previous subscriptions got updated accordingly.

What I did notice is that when I deleted the default card which was associated with a subscription I got this message:

<img width="1131" alt="Screenshot 2023-10-11 at 11 39 33 AM" src="https://github.com/Automattic/woocommerce-payments/assets/6342964/32c621bb-286d-415d-a8cc-74bc15af1cb0">


But notice it says it turned the 4242 card as the default while I actually deleted the 4242 and it was supposed to use the 4444 card as the default instead. 


Closing this issue and creating a new one for the new found bug
